hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileBaseImpl.java

```diff
@@ -88,6 +88,15 @@ protected abstract <T extends BaseRecord> BufferedReader getReader(
   protected abstract <T extends BaseRecord> BufferedWriter getWriter(
       String path);
 
+  /**
+   * Convenience method to allow to mocking of protected
+   * {@link #getWriter(String)}
+   */
+  @VisibleForTesting
+  public BufferedWriter getBufferedWriter(String path) {
```

Comment:
Private? Protected? There is no reason to have `VisibleForTesting` if public.

Reply:
The annotation is to state that the member is only public for tests. So non-test use-cases outside of the class should throw an error.
https://github.com/apache/hadoop/blob/8f971b0e5413b491a2c7043bd25b046777e07395/hadoop-common-project/hadoop-annotations/src/main/java/org/apache/hadoop/classification/VisibleForTesting.java#L27-L38